pds4indextools/pds4_create_xml_index.py

```diff
@@ -434,26 +532,50 @@ def main():
         namespaces['pds'] = namespaces.pop(None)
         prefixes = {v: k for k, v in namespaces.items()}
 
-        xml_results = {}
-        traverse_and_store(root, tree, xml_results, elements_to_scrape,
+        label_results = {}
+        traverse_and_store(root, tree, label_results, elements_to_scrape,
                            nillable_elements_info, config, file)
 
-        for key in list(xml_results.keys()):
-            process_tags(xml_results, key, root,
-                         namespaces, prefixes, args)
+        for key in list(label_results.keys()):
+            process_tags(label_results, key, root,
+                         namespaces, prefixes)
+            
+        new_xpaths = renumber_xpaths(label_results.keys())
+        for key, value in new_xpaths.items():
+            label_results[value] = label_results.pop(key)
+
+        if args.disambiguate_xpaths:
+                elements = ()
+                xpath_elements = []
+                tags = []
+                for key in list(label_results.keys()):
+                    xpath_elements.append(grab_elements(key))
+                
+                duplicates = [t for t in set(xpath_elements) if xpath_elements.count(t) > 1]
+
+                for key in list(label_results.keys()):
+                    elements = grab_elements(key)
+                    tag = elements[-1]
+                    if elements not in duplicates and elements[-1] not in tags:
+                        value = tag
+                        tags.append(tag)
+                    else:
+                        value = key
+                    label_results[value] = label_results.pop(key)
```

Comment:
Doesn't this technique change the order of the columns? Any tag that is unique will end up being at the very end of the list, right?

Reply:
So, it puts the new result at the end of the dictionary. However, since it does this for ALL entries, it keeps putting the next new entry at the end, until all the entries are replaced. It's like a conveyer belt:

a-b-c-d 

b-c-d-a*

c-d-a*-b*

d-a*-b*-c*

a*-b*-c*-d*

Reply:
I saw this above, but for some reason while skimming I thought this code was only executed for some of the labels, resulting in them being out of order. But I see it is done for all labels so it works as expected.